test/cluster_test.go

```diff
@@ -331,7 +336,176 @@ func DeployClusterWithSyslog(ctx goctx.Context) {
+
+					By("Changing ca-file to ca-path in TLS configuration")
+					aeroCluster, err = getCluster(
+						k8sClient, ctx, clusterNamespacedName,
+					)
+					Expect(err).ToNot(HaveOccurred())
+
+					network = aeroCluster.Spec.AerospikeConfig.Value["network"].(map[string]interface{})
+					tlsList = network["tls"].([]interface{})
+					usedTLS := tlsList[0].(map[string]interface{})
+					usedTLS["ca-path"] = "/etc/aerospike/secret/cacerts"
+					delete(usedTLS, "ca-file")
+					tlsList[0] = usedTLS
+					network["tls"] = tlsList
+					aeroCluster.Spec.AerospikeConfig.Value["network"] = network
+					secretVolume := asdbv1.VolumeSpec{
+						Name: tlsCacertSecretName,
+						Source: asdbv1.VolumeSource{
+							Secret: &v1.SecretVolumeSource{
+								SecretName: tlsCacertSecretName,
+							},
+						},
+						Aerospike: &asdbv1.AerospikeServerVolumeAttachment{
+							Path: "/etc/aerospike/secret/cacerts",
+						},
+					}
+					aeroCluster.Spec.Storage.Volumes = append(aeroCluster.Spec.Storage.Volumes, secretVolume)
+					operatorClientCertSpec := getOperatorCert()
+					operatorClientCertSpec.AerospikeOperatorCertSource.SecretCertSource.CaCertsFilename = ""
+					cacertPath := &asdbv1.CaCertsSource{
+						SecretName: tlsCacertSecretName,
+					}
+					operatorClientCertSpec.AerospikeOperatorCertSource.SecretCertSource.CaCertsSource = cacertPath
+					aeroCluster.Spec.OperatorClientCertSpec = operatorClientCertSpec
+					err = updateCluster(k8sClient, ctx, aeroCluster)
+					Expect(err).ToNot(HaveOccurred())
+				},
+			)
+		},
+	)
+
+	Context(
+		"When doing invalid operations", func() {
+			It(
+				"Try update operations", func() {
+					By("Modifying name of used TLS configuration")
+					aeroCluster, err := getCluster(
+						k8sClient, ctx, clusterNamespacedName,
+					)
+					Expect(err).ToNot(HaveOccurred())
+
+					network := aeroCluster.Spec.AerospikeConfig.Value["network"].(map[string]interface{})
+					tlsList := network["tls"].([]interface{})
+					usedTLS := tlsList[0].(map[string]interface{})
+					usedTLS["name"] = "aerospike-a-0.test-runner2"
+					tlsList[0] = usedTLS
+					network["tls"] = tlsList
+					aeroCluster.Spec.AerospikeConfig.Value["network"] = network
+					err = updateCluster(k8sClient, ctx, aeroCluster)
+					Expect(err).Should(HaveOccurred())
+
+					By("Modifying ca-file of used TLS configuration")
+					aeroCluster, err = getCluster(
+						k8sClient, ctx, clusterNamespacedName,
+					)
+					Expect(err).ToNot(HaveOccurred())
+
+					network = aeroCluster.Spec.AerospikeConfig.Value["network"].(map[string]interface{})
+					tlsList = network["tls"].([]interface{})
+					usedTLS = tlsList[0].(map[string]interface{})
+					usedTLS["ca-file"] = "/etc/aerospike/secret/fb_cert.pem"
+					tlsList[0] = usedTLS
+					network["tls"] = tlsList
+					aeroCluster.Spec.AerospikeConfig.Value["network"] = network
+					err = updateCluster(k8sClient, ctx, aeroCluster)
+					Expect(err).Should(HaveOccurred())
+				},
```

Comment:
Should we add steps to check when both ca-file and ca-path are updated in cluster?